ccm_web/server/src/api/api.service.ts

```diff
@@ -138,41 +140,49 @@ export class APIService {
     const adminHandler = new AdminApiHandler(adminRequestor)
     const newUserAccountID = this.configService.get('canvas.newUserAccountID', { infer: true })
 
-    const resultData: ExternalUserData = {}
-    const createErrors: APIErrorData[] = []
+    const userResults: ExternalUserData[] = []
+    const createErrors: APIErrorPayload[] = []
     const newUserEmails: string[] = []
     const createUserResponses = await adminHandler.createExternalUsers(externalUsers, newUserAccountID)
 
     // Handle create user responses: success, failure, and already exists
     createUserResponses.forEach(({ email, result }) => {
+      let userCreated: boolean | APIErrorPayload
       if (isAPIErrorData(result)) {
-        createErrors.push(result)
-      } else if (result !== false) {
+        if (result.errors.length !== 1) {
+          throw Error(`Expected one error payload from Canvas (found ${result.errors.length})`)
+        }
```

Comment:
Were there cases where more than one payload was returned?  Why was that?

Reply:
Honestly, no, I'm just handling an edge case explicitly. I was trying to simplify the error data returned from the endpoint to save the client some processing. Basically we're always building an `APIErrorData` object for Canvas with an array of payloads, but in many cases we only expect one. Here, we expect one for each response, so I'm just providing the payloads in the error response, rather than `APIErrorData`, which was designed to be returned from the controllers. Because the upstream endpoint returns a different shape of data, I'm just extracting the part I need. Sorry, hope that was clear. Let me know if you disagree with any of this.

Reply:
I'm going to merge, but let me know what you think about this. We could open a new issue to investigate how best to handle error data construction in general (we have discussed this before).

Reply:
Please go ahead and merge, We can open separate issue if that is needed.